rewrite-java/src/main/java/org/openrewrite/java/ChangeType.java

```diff
@@ -527,6 +551,25 @@ private boolean hasNoConflictingImport(@Nullable JavaSourceFile sf) {
         }
     }
 
+    @Value
+    @EqualsAndHashCode(callSuper = false)
+    private static class ReferenceChangeTypeVisitor extends TreeVisitor<Tree, ExecutionContext> {
+        Set<Reference> matches;
+        Reference.Renamer renamer;
+        String newFullyQualifiedName;
+
+        @Override
+        public @Nullable Tree preVisit(@Nullable Tree tree, ExecutionContext ctx) {
+            Tree tree1 = super.visit(tree, ctx);
```

Comment:
This call looks suspicious to me, as it calls `visit()` rather than `preVisit()`. I don't think you need to call `super.preVisit()` anyway. You can just return `tree` in the else case.

Reply:
I don't really know what happened here but the tests were failing because of mismatching super calls. Removed the super calls now!